### ISSUE TITLE
composition of maps

### ISSUE THREAD

**Issue by esheldon** [LSSTDESC/healsparse#21]
For map generation, it will be important to have composable maps.

For example, some regions may be flagged in various ways.  We want to be able to bitwise `or` such masks.  Say we have a mask for bright stars and a mask for bad pixels

```python
map_total = map_stars | map_pixels
```

We may also want to compose maps based on some kind of weight. This is useful when creating a depth map for the survey
```python
depth_map_total = sum([map*w for map,w in zip(maps, weights)])
```

I think there are other types of composition that are also useful, they just are not yet coming to mind.

**Comment by erykoff** [comment]
@esheldon So if I were to add to maps together that have slightly different footprints, should the resulting summation only cover those pixels where both were defined, or treat empty pixels as 0s?

**Comment by esheldon** [comment]
I can imagine an example where we want to describe the depth of the survey, in which case we want the summed map to be a union of the pixels in the individual maps, and the summation in each pixel is a kind of weighted sum.

I can also imagine one would want to keep the intersection of many maps, for example if one wanted to describe a coadd region created from the regions describing many single-epoch CCD images.

**Comment by esheldon** [comment]
Maybe something like this
```python
healsparse.sum_union(maps)
healsparse.sum_intersection(maps)
```
I'm not sure if the weights should be in the maps themselves or extra arguments to the sum functions.   Maybe the maps could have weights for each pixel, but there could also be weights sent to the sum functions to define relative weights for the maps.

The example I gave a above `sum([map*w for map,w in zip(maps, weights)])` might be syntactic sugar for either a union or intersection?  Or maybe it is too ambiguous so we should avoid it?

**Comment by erykoff** [comment]
Closed with #22
